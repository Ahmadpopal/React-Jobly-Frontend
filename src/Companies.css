.Companies-div-title {
  margin: 10px;
}

.Companies-title {
  margin: 10px;
  text-align: center;
}
