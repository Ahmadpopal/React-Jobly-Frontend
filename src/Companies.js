import React, { useState, useEffect } from 'react';
import { Container, Row, Col } from 'reactstrap';


import JoblyApi from './Jobly'
import CompanyCard from './CompanyCard'
import SearchForm from './helper/SearchForm'
import './Companies.css'


// Companies Component: 
//   - Show Companies list from Jobly Api 
//   - show Search Form where Users can search companies by passing company name 

// Renders 
//   - SearchForm Component 
//   - CompanyCard Component

const Companies = () => {

    const [companies, setCompanies] = useState(null);

    useEffect(() => {
            getAllCompanies();
    }, []);

    async function getAllCompanies(name) {
      let companies = await JoblyApi.getCompanies(name);
      setCompanies(companies);
    };

    if (!companies) {
      return <h2>Companies List Loading....</h2>;
	}


    return (
      <Container className="Companies">
        <Row>
          <Col className="Companies-div-title">
            <h3 className="Companies-title">Companies list <hr /></h3>
          </Col>
        </Row>
        <Row>
          <Col>
            <SearchForm searchFor={getAllCompanies} />
          </Col>
        </Row>
        <Row>
          <Col>
          {companies.map(c => (
                <CompanyCard key={c.handle} 
                  name={c.name} handle={c.handle} 
                  numOfEmployee={c.numEmployees} 
                  description={c.description} 
                  logoUrl={c.logoUrl} />
              ))}
          </Col>
        </Row>
      </Container>
    )
}


export default Companies