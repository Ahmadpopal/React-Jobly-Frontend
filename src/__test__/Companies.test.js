import React from 'react';
import { render } from '@testing-library/react';

import { MemoryRouter, Route } from 'react-router-dom';
import Companies from '../Companies';

describe('COMPANIES TEST', () => {
	it('Companies Component should render without crashing', () => {
		render(
			<MemoryRouter initialEntries={[ '/companies/' ]}>
				<Route exact path="/companies/">
					<Companies />
				</Route>
			</MemoryRouter>
		);
	});

	it('Companies Snapshot should matches', () => {
		const { asFragment } = render(
			<MemoryRouter initialEntries={[ '/companies/' ]}>
				<Route exact path="/companies/">
					<Companies />
				</Route>
			</MemoryRouter>
		);
		expect(asFragment()).toMatchSnapshot();
	});
});