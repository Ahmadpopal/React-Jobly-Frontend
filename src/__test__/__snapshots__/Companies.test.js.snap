// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPANIES TEST Companies Snapshot should matches 1`] = `
<DocumentFragment>
  <h2>
    Companies List Loading....
  </h2>
</DocumentFragment>
`;
